src/main/java/org/breedinginsight/brapi/v2/services/BrAPIGermplasmService.java

```diff
@@ -149,8 +131,37 @@ public DownloadFile exportGermplasmList(UUID programId, String listId) throws Ap
 
             processedData.add(row);
         }
+        return processedData;
+    }
+
+    public DownloadFile exportGermplasmList(UUID programId, String listId, FileType fileExtension) throws ApiException, IOException {
+        List<Column> columns = GermplasmFileColumns.getOrderedColumns();
+
+        //Retrieve germplasm list data
+        BrAPIListDetails listData = brAPIListDAO.getListById(listId, programId).getResult();
+
+        //Retrieve germplasm data
+        List<String> germplasmNames = listData.getData();
+        List<BrAPIGermplasm> germplasm = germplasmDAO.getGermplasmByRawName(germplasmNames, programId);
+        //processGermplasmForDisplay, numbers
+        germplasm.sort(Comparator.comparingInt(g -> g.getAdditionalInfo().get("importEntryNumber").getAsInt()));
+
+        String listName = listData.getListName();
+        Optional<Program> optionalProgram = programService.getById(programId);
+        if (optionalProgram.isPresent()) {
+            Program program = optionalProgram.get();
+            listName = removeAppendedKey(listName, program.getKey());
+        }
+        String fileName = createFileName(listData, listName);
+        StreamedFile downloadFile;
+        //Convert list data to List<Map<String, Object>> data to pass into file writer
+        List<Map<String, Object>> processedData =  processData(germplasm);
 
-        StreamedFile downloadFile = ExcelWriter.writeToDownload("Germplasm Import", columns, processedData);
+        if (fileExtension.getName().equals("csv")){
```

Comment:
This comparison should be enum comparison: 

`if(fileExtension == FileType.CSV){`

Reply:
fixed